sympy/polys/domains/finitefield.py

```diff
@@ -43,10 +43,11 @@ def _modular_int_factory(mod, dom, symmetric, self):
             flint.nmod(0, mod)
         except OverflowError:
             # Use fmpz_mod
-            ctx = flint.fmpz_mod_ctx(mod)
+            fctx = flint.fmpz_mod_ctx(mod)
+            ctx = lambda x: fctx(int(x))
```

Comment:
This will accept other things like floats:
```python
In [1]: FF(11)(3.5)
Out[1]: 3
```
It should use `operator.index` rather than `int`.

Reply:
Right, will fix that, thanks.

Any other notes? What was the exact behavior before the flint change? What was accepted?

Reply:
If you set the environment variable `SYMPY_GROUND_TYPES=python` then you can test what happens when not using Flint which should be mostly similar to how things were in previous versions of Python although some changes were needed.

The old behaviour is to convert with `FF(11).convert(obj)` which is perhaps more generous about what it accepts.

I don't see why anything that is not coercible to int should be accepted here though regardless of what previous behaviour was.